Makefile

```diff
@@ -2,8 +2,8 @@
 ROOTDIR=$(dir $(abspath $(lastword $(MAKEFILE_LIST))))
 
 # Used to populate version variable in main package.
-VERSION=$(shell git describe --match 'v[0-9]*' --dirty='.m' --always)
-REVISION=$(shell git rev-parse HEAD)$(shell if ! git diff --no-ext-diff --quiet --exit-code; then echo .m; fi)
+VERSION ?= $(shell git describe --match 'v[0-9]*' --dirty='.m' --always)
+REVISION ?= $(shell git rev-parse HEAD)$(shell if ! git diff --no-ext-diff --quiet --exit-code; then echo .m; fi)
```

Comment:
For my own understanding could you please link to bash doc or any page that describes that `?=` means use variable value if provided otherwise calculate from command. I only see `${VAR:-20}` when searching online. I may very well be dumb in looking.

Reply:
In this case it's a makefile syntax; https://www.gnu.org/software/make/manual/html_node/Setting.html#Setting

Reply:
Sure. TL;DR `?=` assigns the expression to the variable if and only if the variable is undefined in the environment. If defined and empty, `?=` will not assign the expression. The expression is lazily evaluated recursively on use, similar to `=`.

[`?=` docs](https://www.gnu.org/software/make/manual/html_node/Setting.html)

It's a surprisingly lesser known feature but extremely useful. Handles cases where you wish for `VERSION` or `REVISION` to be intentionally blanked out.

Reply:
🤦‍♂ Its a makefile change. Thanks for the doc!